src/group.rs

```diff
@@ -32,28 +35,49 @@ impl GroupCreateOpts {
     /// - `add_as_member`
     ///     - `true` - The creating user should be added as a member (in addition to being a group admin)
     ///     - `false` - The creating user will not be a member of the group, but will still be an admin.
+    /// - `members` - List of users to be added as members of the group.
+    ///               Note: even if `add_as_member` is false, the calling user will be added as a member if they are in this list.
     /// - `needs_rotation`
     ///     - true - group's private key will be marked for rotation
     ///     - false (default) - group's private key will not be marked for rotation
     pub fn new(
         id: Option<GroupId>,
         name: Option<GroupName>,
         add_as_member: bool,
+        members: Vec<UserId>,
         needs_rotation: bool,
     ) -> GroupCreateOpts {
         GroupCreateOpts {
             id,
             name,
             add_as_member,
+            members,
             needs_rotation,
         }
     }
+
+    fn standardize(self, calling_id: &UserId) -> GroupCreateOpts {
+        let standardized_members = if self.add_as_member && !self.members.contains(calling_id) {
+            let mut members = self.members.clone();
+            members.push(calling_id.clone());
+            members
+        } else {
+            self.members
+        };
+        GroupCreateOpts::new(
+            self.id,
+            self.name,
+            self.add_as_member,
+            standardized_members,
+            self.needs_rotation,
+        )
+    }
 }
 
 impl Default for GroupCreateOpts {
     fn default() -> Self {
         // membership is the default!
-        GroupCreateOpts::new(None, None, true, false)
+        GroupCreateOpts::new(None, None, true, Vec::new(), false)
```

Comment:
I would suggest creating a new vec with `vec![]`
Applicable in some other places in this PR as well...